### DISCUSSION TITLE
Documenting available internal templates

### DISCUSSION THREAD

**Discussion by bkanuka** [argoproj/argo-workflows#12743] [General]
At my company we have slowly created a number of helpful, reusable templates. These are cluster-level templates that are applicable to a number of different projects.  Ideally these templates are used whenever possible to reduce duplication.  However, I find that they are little difficult to find.  For example, there is a "description" for each input parameter, but these are not easy to find in the Argo UI.

How do you share templates internally?  Do you simply document them separately in [company wiki]?  In my dream world, the template YAML itself could be parsed into a searchable "library" of templates.

**Comment by wesleyscholl** [comment]
@bkanuka 

At my company we all have access to shared cluster workflow templates. Some are documented, some are not. I have a series of articles with code examples and templates on our Confluence. Other engineers have shared similar Argo Workflow templates for their area of expertise. Technically all of our yaml is indexed and is searchable in GitHub. Most of our engineers either find what they need searching internally on Confluence, using cluster workflow templates, searching GitHub, or asking the team if anyone else has built something similar. It's nice to hear from others doing similar things using Argo Workflows.
